src/ilk-deployer.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+/// ilkfab.sol -- Non-standard token adapters
+
+// Copyright (C) 2018 Rain <rainbreak@riseup.net>
+// Copyright (C) 2018-2020 Maker Ecosystem Growth Holdings, INC.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.12;
+
+import "./join.sol";
+import "./join-auth.sol";
+import "./join-2.sol";
+import "./join-3.sol";
+import "./join-4.sol";
+import "./join-5.sol";
+import "./join-6.sol";
+import "./join-7.sol";
+import "./join-8.sol";
+
+interface Changelog {
+    function getAddress(bytes32) external view returns (address);
+}
+
+interface Auth {
+    function rely(address) external;
+    function deny(address) external;
+}
+
+interface FlipFab {
+    function newFlip(address,address,bytes32) external returns (address);
+}
+
+contract IlkDeployer {
```

Comment:
If we are doing this approach of using the chainlog and having to have tests that require a mainnet node I'd suggest all this is moved to a new repo that imports `dss-gem-joins` as a dependency, so we avoid when running the tests for it having this mainnet dependency. E.g. the scripts will suffer if we enable the tests of the dependencies as these ones will fail.